host-bin/edit-chroot

```diff
@@ -385,8 +432,22 @@ for NAME in "$@"; do
         fi
         echo -n "  Restoring $src to $CHROOT..." 1>&2
         mkdir -p "$CHROOT"
+        if [ -f "$src.part01" ]; then
+            tarsrc="`mktemp -u --tmpdir=/tmp 'crouton-restore.XXX'`"
+            mkfifo -m 600 "$tarsrc"
+            # Pipe through dd to eat the EOFs
+            cat "$src" "$src.part"* | dd status=none >> "$tarsrc" &
+            catpid="$!"
+            addtrap "rm -f '$tarsrc'; kill '$catpid' 2>/dev/null"
+        else
+            tarsrc="$src"
+            catpid=''
+        fi
         spinner 1 tar --checkpoint=200 --checkpoint-action=exec=echo \
-            --one-file-system -xaf "$src" -C "$CHROOT" --strip-components=1
+            --one-file-system -xaf "$tarsrc" -C "$CHROOT" --strip-components=1
```

Comment:
Actually, I'm really unclear how this works. I didn't try your modified `edit-chroot`, but I ran some small test on Arch, and Chrome OS (peppy beta, `6457.36.0`), and somethign like `cat file.tar.gz > fifo`, `tar xavf fifo` does not work (tar hangs). 

Looking at `ltrace` output, `tar` (when `-a` is specified), first reads a block from the file, then closes it, and `gzip` then reopens the file. That won't work with fifos. We need to detect the archive type (`-z`, `-j`).